src/integration/UsesTmpDir.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Flarum\Testing\integration;
+
+trait UsesTmpDir
+{
+    protected $DEFAULT_TMP_DIR = __DIR__.'/tmp';
+
+    public function tmpDir() {
+        return realpath(getenv('FLARUM_TEST_TMP_DIR_LOCAL') ?: getenv('FLARUM_TEST_TMP_DIR') ?: static::$DEFAULT_TMP_DIR);
```

Comment:
> Aren't we supposed to use the current working directory path with one of the variables ?

Doesn't `static::$DEFAULT_TMP_DIR` do that already?

> Also, why two variables instead of one ?

This way, a local config var can be set independently of a global config var.

Reply:
> This way, a local config var can be set independently of a global config var.
I see

> Doesn't `static::$DEFAULT_TMP_DIR` do that already?

Not sure if we're talking about the same thing then, that property's value will always be the path `vendor/flarum/testing/src/integration/tmp`. But it doesn't matter either way, I was actually wrong, we don't need to use getcwd with the env variables, because using a relative path value will work just fine.

That said, the property isn't static but you're calling it statically, but since it's only used in one spot, I'd rather it be hardcoded tbh, just to avoid the uppercase variable to look like a constant but isn't actually one issue :p

Reply:
> that property's value will always be the path vendor/flarum/testing/src/integration/tmp

Well, yes. I thought the plan from https://github.com/flarum/core/issues/2544 was to use the vendor location for the default tmp dir, to avoid making assumptions about how an extension might decide to structure its code.

> That said, the property isn't static but you're calling it statically, but since it's only used in one spot

Done, since it's only used in one place in one class, I think hardcoding is fine here.

Reply:
yup that was always the plan, I just got confused and forgot that with the env variables we can also just use relative paths, since such a path will be relative to the current working directory.